lms/djangoapps/mfe_config_api/views.py

```diff
@@ -60,12 +69,50 @@ def get(self, request):
         if not settings.ENABLE_MFE_CONFIG_API:
             return HttpResponseNotFound()
 
-        mfe_config = configuration_helpers.get_value('MFE_CONFIG', settings.MFE_CONFIG)
-        if request.query_params.get('mfe'):
-            mfe = str(request.query_params.get('mfe'))
+        # Get values from django settings (level 6) or site configuration (level 5)
+        base_config = self._get_base_config()
+
+        # Get values from mfe configuration, either from django settings (level 4) or site configuration (level 3)
+        mfe_config = configuration_helpers.get_value("MFE_CONFIG", settings.MFE_CONFIG)
+
+        # Get values from mfe overrides, either from django settings (level 2) or site configuration (level 1)
+        mfe_config_overrides = {}
+        if request.query_params.get("mfe"):
+            mfe = str(request.query_params.get("mfe"))
             app_config = configuration_helpers.get_value(
-                'MFE_CONFIG_OVERRIDES',
+                "MFE_CONFIG_OVERRIDES",
                 settings.MFE_CONFIG_OVERRIDES,
             )
-            mfe_config.update(app_config.get(mfe, {}))
-        return JsonResponse(mfe_config, status=status.HTTP_200_OK)
+            mfe_config_overrides = app_config.get(mfe, {})
+
+        # Merge the three configs in the order of precedence
+        merged_config = base_config | mfe_config | mfe_config_overrides
+
+        return JsonResponse(merged_config, status=status.HTTP_200_OK)
+
+    @staticmethod
+    def _get_base_config() -> dict:
```

Comment:
```suggestion
    def _get_legacy_config() -> dict:
```

You'll need to update the calls as well but these are specifically a compatibility layer that is meant to be temporary, the naming should indicate that as should the docstring.

Reply:
Done